tests/unit/multiclass/KNN_unittest.cc

```diff
@@ -226,3 +226,79 @@ TEST(KNN, classify_multiple_kdtree)
 
 
 }
+
+
+TEST(KNN, nearest_neighbours_brute)
+{
+	//simple non-trivial test
+	//initialize labels
+	SGVector<float64_t> lab({0.0, 0.0, 1.0, 1.0, 0.0});
+	//initialize features
+	SGMatrix<float64_t> feat({{2.0, 1.0}, {3.0, 1.0}, {6.0, 1.0}, {7.0, 1.0}, {3.5, 2.0}});
+	//divide training and testing parts
+	SGVector<index_t> train ({0, 1, 2, 3});
+	SGVector<index_t> test ({4});
+
+	auto labels = std::make_shared<MulticlassLabels>(lab);
+	auto features = std::make_shared<DenseFeatures<float64_t> >(feat);
+	auto features_test = features->clone()->as<DotFeatures>();
+	auto labels_test = labels->clone()->as<MulticlassLabels>();
+
+	int32_t k=3;
+	auto distance = std::make_shared<EuclideanDistance>();
+	auto knn=std::make_shared<KNN> (k, distance, labels, KNN_BRUTE);
+
+	features->add_subset(train);
+	labels->add_subset(train);
+	knn->train(features);
+
+	features_test->add_subset(test);
```

Comment:
I see.
Let's not use this method, as it is deprecated. There is a `view` function somewhere, that creates a new instance using the name underlying memory

Reply:
Done.